internal/controlplane/handlers_oauth.go

```diff
@@ -115,108 +126,150 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 	}, nil
 }
 
-// ExchangeCodeForTokenCLI exchanges an OAuth2 code for a token
-// This function gathers the state from the database and compares it to the state
-// passed in. If they match, the code is exchanged for a token.
-// This function is used by the CLI client.
-func (s *Server) ExchangeCodeForTokenCLI(ctx context.Context,
-	in *pb.ExchangeCodeForTokenCLIRequest) (*httpbody.HttpBody, error) {
-
-	// Configure tracing
-	span := trace.SpanFromContext(ctx)
-	span.SetName("server.ExchangeCodeForTokenCLI")
-	span.SetAttributes(attribute.Key("code").String(in.Code))
-	defer span.End()
-
-	// Check the nonce to make sure it's valid
-	valid, err := mcrypto.IsNonceValid(in.State, s.cfg.Auth.NoncePeriod)
-
-	if err != nil {
-		return nil, status.Errorf(codes.Unknown, "error checking nonce: %s", err)
-	}
+// HandleProviderCallback handles the OAuth 2.0 authorization code callback from the enrolled
+// provider. This function gathers the state from the database and compares it to the state
+// passed in. If they match, the provider code is exchanged for a provider token.
+//
+//nolint:gocyclo
```

Comment:
maybe we should listen to this warning and address the cyclomatic complexity by splitting this function.

Reply:
Done